snippets/sdk/python/certificates/create_or_update.py

```diff
@@ -0,0 +1,180 @@
+#! /usr/local/env python3
+
+import os
+
+from azure.keyvault.certificates import (
+    CertificateClient,
+    CertificatePolicy,
+)
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.authorization import AuthorizationManagementClient
+from azure.mgmt.msi import ManagedServiceIdentityClient
+from azure.mgmt.keyvault import KeyVaultManagementClient
+from azure.mgmt.nginx import NginxManagementClient
+
+
+def main():
+    TENANT_ID = os.environ.get("AZURE_TENANT_ID", None)
+    SUBSCRIPTION_ID = os.environ.get("AZURE_SUBSCRIPTION_ID", None)
+    GROUP_NAME = "myResourceGroup"
+    DEPLOYMENT_NAME = "myDeployment"
+    VAULT_NAME = "myVault123456"
+    KV_URL = "https://" + VAULT_NAME + ".vault.azure.net"
+    CERTIFICATE_NAME = "myCert"
+    MANAGED_IDENTITY_NAME = "myIdentity"
+
+    # Create clients
+    # For other authentication approaches, please see: https://pypi.org/project/azure-identity/
+    auth_client = AuthorizationManagementClient(
+        credential=DefaultAzureCredential(), subscription_id=SUBSCRIPTION_ID
+    )
+    msi_client = ManagedServiceIdentityClient(
+        credential=DefaultAzureCredential(), subscription_id=SUBSCRIPTION_ID
+    )
+    keyvault_client = KeyVaultManagementClient(
+        credential=DefaultAzureCredential(), subscription_id=SUBSCRIPTION_ID
+    )
+    certificate_client = CertificateClient(
+        credential=DefaultAzureCredential(),
+        vault_url=KV_URL,
+    )
+    nginx_client = NginxManagementClient(
+        credential=DefaultAzureCredential(),
+        subscription_id=SUBSCRIPTION_ID,
+    )
+
+    # Fetch our identity
+    identity = msi_client.user_assigned_identities.get(
+        GROUP_NAME, MANAGED_IDENTITY_NAME
+    )
+
+    # Create vault
+    # https://github.com/Azure-Samples/azure-samples-python-management/tree/main/samples/keyvault
+    # A managed identity only requires the 'Key Vault Secrets User' role to give NGINXaaS for Azure
+    # the appropriate permissions to pull certificates from AKV
+    vault = keyvault_client.vaults.begin_create_or_update(
+        GROUP_NAME,
+        VAULT_NAME,
+        {
+            "location": "eastus",
+            "properties": {
+                "tenant_id": TENANT_ID,
+                "sku": {"family": "A", "name": "standard"},
+                "access_policies": [],
+                "enable_rbac_authorization": True,
+                "enabled_for_deployment": True,
+                "enabled_for_disk_encryption": True,
+                "enabled_for_template_deployment": True,
+            },
+        },
+    ).result()
+    print("Created a vault:\n{}".format(vault))
+
+    # Give current user key vault administrator role assignment
+    role_id = "00482a5a-887f-4fb3-b363-3b7fe8e74483"
```

Comment:
can we use names instead of magic uuids?

```suggestion
    role_id = "Key Vault Administrator"
```

Reply:
Sadly nope we have to use the magic UUIDs. I can try and find the SDK constant that refers to it. Everywhere I have seen role assignment API docs it has used these magic strings and pointed to the official docs for folks who are interested in more role assignment role IDs. I tried to document what that UUID was and where to find all of them consistently with what MSFT docs and SDK snippets use.